vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleRegistrationSpreadsheet.java

```diff
@@ -398,42 +405,78 @@ public List<SamplesheetHeaderName> retrieveGenomics() {
+      String specimenInput = parseAndTrimCellOfRow(row, header.indexOf(SamplesheetHeaderName.SPECIMEN));
+      String analyteInput = parseAndTrimCellOfRow(row, header.indexOf(SamplesheetHeaderName.ANALYTE));
+      String commentInput = parseAndTrimCellOfRow(row, header.indexOf(SamplesheetHeaderName.CUSTOMER_COMMENT));
+
+      // we need to stream this list twice, so we use a supplier
+      Supplier<Stream<String>> mandatoryCellStreamSupplier = () -> Stream.of(analysisTypeInput,
+          sampleLabelInput,
+          replicateIDInput, conditionInput, speciesInput, specimenInput, analyteInput);
 
       if (mandatoryCellStreamSupplier.get().anyMatch(Objects::isNull)) {
         break;
       }
-      if (mandatoryCellStreamSupplier.get().noneMatch(x -> x.getStringCellValue().isEmpty())) {
-        ExperimentalGroup experimentalGroup = experimentalGroupToConditionString.get(
-            conditionCell.getStringCellValue().trim());
-        Long experimentalGroupId = experimentalGroup.id();
-        BiologicalReplicateId biologicalReplicateId = retrieveBiologicalReplicateId(
-            replicateIDCell.getStringCellValue().trim(), conditionCell.getStringCellValue().trim());
-        rows.add(new NGSRowDTO(analysisTypeCell.getStringCellValue().trim(),
-            sampleLabelCell.getStringCellValue().trim(),
-            biologicalReplicateId, experimentalGroupId,
-            speciesCell.getStringCellValue().trim(), specimenCell.getStringCellValue().trim(),
-            analyteCell.getStringCellValue().trim(),
-            commentCell.getStringCellValue().trim()));
+
+      if(mandatoryCellStreamSupplier.get().noneMatch(x -> x.isEmpty())) {
+        String uniqueSampleString = replicateIDInput+conditionInput;
+        if(uniqueSamples.contains(uniqueSampleString)) {
+          return Result.fromError(new SpreadsheetValidationException(
+              "Biological replicate Id "+replicateIDInput+" was used multiple times for the same condition.", rowId));
+        } else {
+          ExperimentalGroup experimentalGroup = experimentalGroupToConditionString.get(
+              conditionInput);
+          Long experimentalGroupId = experimentalGroup.id();
+          BiologicalReplicateId biologicalReplicateId = retrieveBiologicalReplicateId(
+              replicateIDInput, conditionInput);
+          rows.add(new NGSRowDTO(analysisTypeInput, sampleLabelInput,
+              biologicalReplicateId, experimentalGroupId,
+              speciesInput, specimenInput, analyteInput,
+              commentInput));
+          uniqueSamples.add(uniqueSampleString);
+        }
+      } else {
+        return Result.fromError(new SpreadsheetValidationException(
+            "Missing mandatory fields.", rowId));
+      }
+
+    }
+    return Result.fromValue(rows);
+  }
+
+  private String parseAndTrimCellOfRow(Row row, int colId) {
+    Cell cell = row.getCell(colId);
+    if(cell==null) {
+      return null;
```

Comment:
why do we return null here? maybe an Optional<String> would clarify the expected return value? Or is this unexpected? Then throw an exception an crash! ;)

Reply:
The cell coming in can be null, so in that case null needs to be returned. Otherwise the validation method also gets more complex. I have now added cases for all but one of the other CellTypes (which itself seems to be an error case), so they won't return null any more.